src/Sylius/Bundle/AddressingBundle/Doctrine/ORM/ZoneRepository.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\AddressingBundle\Doctrine\ORM;
+
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+use Sylius\Component\Addressing\Repository\ZoneRepositoryInterface;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+class ZoneRepository extends EntityRepository implements ZoneRepositoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function findOneByName($name)
+    {
+         return $this->createQueryBuilder('o')
+            ->where('o.name = :name')
+            ->setParameter('name', $name)
```

Comment:
The name is not translatable and we don't need to join anything, so I think this method is redundant and we should use simply `findOneBy(array('name' => $name))`.